test/no-package-flags-to-cmj.t

```diff
@@ -0,0 +1,66 @@
+Set up a few directories we'll need
+
+  $ export MELANGELIB="$INSIDE_DUNE/lib/melange"
+  $ mkdir -p lib
+  $ mkdir -p app
+  $ mkdir -p lib/.objs/melange
+  $ mkdir -p app/.objs/melange
+  $ mkdir -p output/lib
+  $ mkdir -p output/app
+
+  $ echo "let t = 1" > lib/a.ml
+  $ echo "let t = A.t" > app/b.ml
+
+Build cmjs
+
+  $ melc -bs-stop-after-cmj lib/a.ml -o lib/.objs/melange/a.cmj
+
+  $ melc -I lib/.objs/melange -bs-stop-after-cmj app/b.ml -o app/.objs/melange/b.cmj
+
+Linking step
+
+  $ melc -bs-package-output lib/ lib/.objs/melange/a.cmj -o output/lib/a.js
+
+If we pass some file extension or module system in `-bs-package-output`, it
```

Comment:
@jchavarri I actually changed this. If you pass `bs-package-output` with package specs Melange will use that. The correct flag to use to control JS output at this stage is `-bs-module-type`

Reply:
Sounds good.

re: passing `-o` with file extension and `-bs-package-output`, I am not sure if we should fail? It's a bit confusing that user can pass file extension in two places.

Reply:
it's indeed a bit confusing, but I think we don't need to care about that:
- the CLI interface is for "expert" usage
- people will mostly interact with melange through `mel` or `dune`

Reply:
I think that's one (maybe last?) thing to improve on later. I think the code is a lot cleaner after #384, as there's now a separation between the "package specs" (as recorded in the `.cmj` -- what's the relative path of this module?) and the output information we'll need for emission ("what module system and JS suffix should this file have?").